storefront/app/controllers/spree/checkout_controller.rb

```diff
@@ -191,6 +190,8 @@ def load_order
             redirect_to spree_login_path
           end
         end
+      elsif !current_store.prefers_guest_checkout?
+        require_user(redirect_path: spree_signup_path)
```

Comment:
@damianlegawiec I moved the `require_user` part with the sign up path inside `load_order`. We're already handling here cases when trying to access a checkout for existing users. 

With this change we'll also redirect guest with new checkouts to the signup page.